DigitalLearningSolutions.Data/Services/NotificationService.cs

```diff
@@ -1,61 +1,81 @@
-            this.configService = configService;
-            this.emailService = emailService;
-        }
-
-        public void SendUnlockRequest(int progressId)
-        {
-            var unlockData = notificationDataService.GetUnlockData(progressId);
-            if (unlockData == null)
-            {
-                throw new NotificationDataException($"No record found when trying to fetch Unlock Data. Progress ID: {progressId}");
-            }
-
-            unlockData.ContactForename = unlockData.ContactForename == "" ? "Colleague" : unlockData.ContactForename;
-            var trackingSystemBaseUrl = configService.GetConfigValue(ConfigService.TrackingSystemBaseUrl) ??
-                                        throw new ConfigValueMissingException(configService.GetConfigValueMissingExceptionMessage("TrackingSystemBaseUrl"));
-            var unlockUrl = new UriBuilder(trackingSystemBaseUrl);
-            unlockUrl.Path += "coursedelegates";
-            unlockUrl.Query = $"CustomisationID={unlockData.CustomisationId}";
-            string emailSubjectLine = "Digital Learning Solutions Progress Unlock Request";
-            var builder = new BodyBuilder
-            {
+﻿namespace DigitalLearningSolutions.Data.Services
+{
+    using System;
+    using DigitalLearningSolutions.Data.Models;
+    using DigitalLearningSolutions.Data.Models.Email;
+    using Microsoft.FeatureManagement;
+    using MimeKit;
+
+    public interface INotificationService
+    {
+        void SendUnlockRequest(int progressId);
+    }
+
+    public class NotificationService : INotificationService
+    {
+        private readonly IConfigService configService;
+        private readonly IEmailService emailService;
+        private readonly IFeatureManager featureManager;
+        private readonly INotificationDataService notificationDataService;
+
+        public NotificationService(
+            INotificationDataService notificationDataService,
+            IConfigService configService,
+            IEmailService emailService,
+            IFeatureManager featureManager
+        )
+        {
+            this.notificationDataService = notificationDataService;
+            this.configService = configService;
+            this.emailService = emailService;
+            this.featureManager = featureManager;
+        }
+
+        public void SendUnlockRequest(int progressId)
+        {
+            var unlockData = notificationDataService.GetUnlockData(progressId);
+            if (unlockData == null)
+            {
+                throw new NotificationDataException(
+                    $"No record found when trying to fetch Unlock Data. Progress ID: {progressId}"
+                );
+            }
+
+            unlockData.ContactForename = unlockData.ContactForename == "" ? "Colleague" : unlockData.ContactForename;
+            var trackingSystemBaseUrl = configService.GetConfigValue(ConfigService.TrackingSystemBaseUrl) ??

```

Comment:
Looking at lines 45 to 58, can we look at simplyfying this? Example could be:

```
var refactoredTrackingSystemEnabled = await featureManager.IsEnabledAsync("RefactoredTrackingSystem");
var baseUrlConfigOption = refactoredTrackingSystemEnabled ? ConfigService.AppBaseUrl : ConfigService.TrackingSystemBaseUrl;
var trackingSystemBaseUrl = configService.GetConfigValue(baseUrlConfigOption) ??
    throw new ConfigValueMissingException(
      configService.GetConfigValueMissingExceptionMessage(baseUrlConfigOption)
    ); 
var unlockUrl = new UriBuilder(trackingSystemBaseUrl);
```

Reply:
I refactored it as suggested but SendUnlockRequest is not async since it was told it is not necessary in show and tell